eis_toolkit/raster_processing/distance_to_anomaly.py

```diff
@@ -0,0 +1,200 @@
+from itertools import chain
+from numbers import Number
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+import geopandas as gpd
+import numpy as np
+import rasterio
+from beartype import beartype
+from beartype.typing import Literal, Tuple, Union
+from rasterio import profiles, transform
+
+from eis_toolkit.exceptions import InvalidParameterValueException
+from eis_toolkit.utilities.miscellaneous import row_points, toggle_gdal_exceptions
+from eis_toolkit.vector_processing.distance_computation import distance_computation
+
+THRESHOLD_CRITERIA_VALUE_TYPE = Union[Tuple[Number, Number], Number]
+THRESHOLD_CRITERIA_TYPE = Literal["lower", "higher", "in_between", "outside"]
+
+
+@beartype
+def distance_to_anomaly(
+    raster_profile: Union[profiles.Profile, dict],
+    anomaly_raster_profile: Union[profiles.Profile, dict],
+    anomaly_raster_data: np.ndarray,
```

Comment:
A question about these: Is `raster_profile` the target/output profile? For me, it was not (and maybe isn't still) completely clear why we need to specify two raster profiles. In what cases `raster_profile` != `anomaly_raster_profile`? If we keep both, the naming/docs could be clarified.

Also, our raster processing functions have taken in Rasterio's DatasetReaders so far and this implementation conflicts that convention. However, I have to say I prefer this way in general, so maybe this does not need to be changed and we can considering refactoring the other tools.

Reply:
With the initial (non-gdal) approach it was possible to calculate the distances for a raster that is not equivalent in profile to the raster in which the anomalies are determined from. So anomaly raster could have e.g. different resolution to the raster in which the distances were calculated to. For simplification sense I have now removed this option, I suppose it is probably not a common use case. Easier to maintain without it 👍